Quaver.Shared/Screens/Result/UI/ResultHitDifferenceGraph.cs

```diff
@@ -0,0 +1,305 @@
+                Alpha = 0.3f,
+                Tint = SkinManager.Skin.Keys[GameMode.Keys4].JudgeColors[smallestJudgementPair.Key],
+                Y = y,
+                Size = new ScalableVector2(Width, middle - y),
+            });
+
+            // Area below the middle line.
+            judgementAreas.Add(new Sprite()
+            {
+                Parent = this,
+                Alpha = 0.3f,
+                Tint = SkinManager.Skin.Keys[GameMode.Keys4].JudgeColors[smallestJudgementPair.Key],
+                Y = middle,
+                Size = new ScalableVector2(Width, (int) convertDifferenceToY(-smallestJudgementPair.Value) - middle),
+            });
+
+            var i = 2;
+            foreach (var (judgement, difference) in Processor.JudgementWindow.OrderBy(x => x.Value).Skip(1))
+            {
+                // Area above the middle line.
+                y = (int) convertDifferenceToY(difference);
+                judgementAreas.Add(new Sprite()
+                {
+                    Parent = this,
+                    Alpha = 0.3f,
+                    Tint = SkinManager.Skin.Keys[GameMode.Keys4].JudgeColors[judgement],
+                    Y = y,
+                    Size = new ScalableVector2(Width, judgementAreas[i - 2].Y - y),
+                });
+
+                // Area below the middle line.
+                y = (int) (judgementAreas[i - 1].Y + judgementAreas[i - 1].Height);
+                judgementAreas.Add(new Sprite()
+                {
+                    Parent = this,
+                    Alpha = 0.3f,
+                    Tint = SkinManager.Skin.Keys[GameMode.Keys4].JudgeColors[judgement],
+                    Y = y,
+                    Size = new ScalableVector2(Width, (int) convertDifferenceToY(-difference) - y),
+                });
+
+                i += 2;
+            }
+        }
+
+        /// <summary>
+        /// Creates the early and late text labels.
+        /// </summary>
+        /// <param name="largestJudgement"></param>
+        private void CreateEarlyLateText(float largestJudgement)
+        {
+            largestJudgement /= ModHelper.GetRateFromMods(Processor.Mods);
+
+            var _ = new SpriteText(Fonts.SourceSansProSemiBold, $"Late (+{largestJudgement}ms)", 13)
+            {
+                Parent = this,
+            };
+
+            var early = new SpriteText(Fonts.SourceSansProSemiBold, $"Early (-{largestJudgement}ms)", 13)
+            {
+                Parent = this,
+            };
+            early.Y = Height - early.Height;
+        }
+
+        /// <summary>
+        /// Filters out only those stats that we care about and computes earliest and latest hit time.
+        /// </summary>
+        private void FilterHitStats()
+        {
+            FilteredStats = new List<HitStat>();
+            EarliestHitTime = int.MaxValue;
+            LatestHitTime = int.MinValue;
+
+            foreach (var breakdown in Processor.Stats.Where(
```

Comment:
For the `Where`, it would be better if you can place this into a method, so it looks a bit cleaner.